app/src/main/java/net/vonforst/evmap/fragment/MapFragment.kt

```diff
@@ -566,29 +566,31 @@ class MapFragment : Fragment(), OnMapReadyCallback, MapsActivity.FragmentCallbac
             }
         })
         vm.chargepoints.observe(viewLifecycleOwner, Observer { res ->
-            when (res.status) {
-                Status.ERROR -> {
-                    val view = view ?: return@Observer
-
-                    connectionErrorSnackbar?.dismiss()
-                    connectionErrorSnackbar = Snackbar
-                        .make(view, R.string.connection_error, Snackbar.LENGTH_INDEFINITE)
-                        .setAction(R.string.retry) {
-                            connectionErrorSnackbar?.dismiss()
-                            vm.reloadChargepoints()
-                        }
-                    connectionErrorSnackbar!!.show()
-                }
-                Status.SUCCESS -> {
-                    connectionErrorSnackbar?.dismiss()
-                }
-                Status.LOADING -> {
-                }
-            }
 
             val chargepoints = res.data
             if (chargepoints != null) {
                 updateMap(chargepoints)
+                when (res.status) {
+                    Status.ERROR -> {
+                        if( chargepoints.size < 1) { //No error complaint if cached are shown.
```

Comment:
`< 1` can be simplified to `== 0` (size is never negative). Also, Kotlin has `.isEmpty()` or `.isNotEmpty()` extension functions as an alternative.

Reply:
It is possible (but not in this case) that it might be 0.001... or unexpected -1, and a <1 would catch any unexpected cases.

I have revised this though with isEmpty.